src/overlays/actors/ovl_Bg_Mori_Elevator/z_bg_mori_elevator.c

```diff
@@ -20,37 +34,247 @@ const ActorInit Bg_Mori_Elevator_InitVars = {
     (ActorFunc)BgMoriElevator_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1800.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A18FC.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+extern UNK_TYPE D_060035F8;
+extern Gfx D_06002AD0[];
+
+f32 func_808A1800(f32* posY, f32 target, f32 arg2, f32 yVel, f32 arg4) {
+    f32 phi_f2;
+
+    phi_f2 = (target - *posY) * arg2;
+    if (*posY < target) {
+        if (yVel < phi_f2) {
+            phi_f2 = yVel;
+        } else {
+            if (phi_f2 < arg4) {
+                phi_f2 = arg4;
+            }
+        }
+        *posY = (*posY + phi_f2);
+
+        if (target < *posY) {
+            *posY = target;
+        }
+    } else {
+        if (target < *posY) {
+            if (phi_f2 < (-yVel)) {
+                phi_f2 = (-yVel);
+            } else {
+                if ((-arg4) < phi_f2) {
+                    phi_f2 = (-arg4);
+                }
+            }
+            *posY = (*posY + phi_f2);
+            if (*posY < target) {
+                *posY = target;
+            }
+        } else {
+            phi_f2 = 0.0f;
+        }
+    }
+    return phi_f2;
+}
+
+void func_808A18FC(BgMoriElevator* this, f32 distTo) {
+    f32 phi_f12;
+    f32 temp_f2;
+    f32 temp1;
+
+
+    temp_f2 = fabsf(distTo) * 0.09f; 
+    if (temp_f2 < 0.0f) {
+    // clang-format off
+        phi_f12 = 0.0f; } else { temp1 = (temp_f2 > 1.0f) ? (1.0f) : (temp_f2);phi_f12 = temp1;} func_800F436C(&this->dyna.actor.projectedPos, NA_SE_EV_ELEVATOR_MOVE2 - SFX_FLAG, phi_f12);
+    // clang-format on
+}
+
+void BgMoriElevator_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgMoriElevator* this = THIS;
+    s32 pad;
+    s32 sp24;
+    sp24 = 0;
+
+    this->unk_172 = sIsSpawned;
+    this->objectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_MORI_TEX);
```

Comment:
This deals specifically with the mori tex object, which is loaded in addition to the object for this actor.
```suggestion
    this->moriTexObjIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_MORI_TEX);
```